src/core/net/srp_server.cpp

```diff
@@ -386,7 +386,7 @@ void Server::CommitSrpUpdate(Error                    aError,
                 Service *newService = existingHost->AddService(service->GetFullName());
 
                 VerifyOrExit(newService != nullptr, aError = kErrorNoBufs);
-                SuccessOrExit(aError = newService->CopyResourcesFrom(*service));
+                newService->TakeResourcesFrom(*service);
                 otLogInfoSrp("[server] %s service %s", (existingService != nullptr) ? "update existing" : "add new",
                              service->GetFullName());
```

Comment:
it may be better for us to stop accessing the service whose resources has been removed, though we don't actually remove its name.
```suggestion
                             newService->GetFullName());
```

Reply:
Good point! Will change it. 

We don't currently move  the `service->mFullname` (and I think copy it in `AddService()` which will call `Service::New(const char *aFullName)`) but we can consider changing it later to do this (would be a bit more efficient). I won't change it from this PR though (to keep the change small).